docs/better-burgers/index.md

```diff
@@ -0,0 +1,345 @@
+  15.  // base cost
+  +. float_of_int(onions)
+  *. 0.2
+  +. float_of_int(cheese)
+  *. 0.1;
+};
+```
+
+You get an error from Melange:
+
+```
+File "src/better-burgers/Item.re", line 24, characters 34-41:
+24 |   let toPrice = ({onions, cheese, lettuce}) => {
+                                       ^^^^^^^
+Error (warning 27 [unused-var-strict]): unused variable lettuce.
+```
+
+Since Madame Jellobutter doesn't want to charge for lettuce, we don't need the
+value of the `lettuce` variable and remove it:
+
+```reason
+let toPrice = ({onions, cheese}) => {
+  15.  // base cost
+  +. float_of_int(onions)
+  *. 0.2
+  +. float_of_int(cheese)
+  *. 0.1;
+};
+```
+
+However, this results in a different error:
+
+```
+File "src/better-burgers/Item.re", line 24, characters 17-33:
+24 |   let toPrice = ({onions, cheese}) => {
+                      ^^^^^^^^^^^^^^^^
+Error (warning 9 [missing-record-field-pattern]): the following labels are not bound in this record pattern:
+lettuce
+Either bind these labels explicitly or add '; _' to the pattern.
+```
+
+OCaml doesn't like that you left the `lettuce` field out of the pattern match.
+We could take the advice of the error message and add a wildcard to the end of
+the record pattern match:
+
+<<< Item.re#record-wildcard
+
+But this makes our code less future-proof. For example, what if you added a
+`bacon` field to the `Burger.t` record and it was expected that adding bacon
+increases the price? With the wildcard in place, the compiler would not tell you
+to update the `Burger.toPrice` function. A better way would be to match on
+`lettuce` but put a wildcard for its value:
+
+<<< Item.re#record-lettuce-wildcard
+
+This tells the compiler that you expect a `lettuce` field but you don't intend
+to use its value inside the function. Similarly to how it's preferable to match
+on all constructors of a variant type, it's also a good idea to match on all
+fields of a record type.
+
+## Pattern matching records
+
+Right now, if a customer doesn't add any toppings to a burger, `Burger.toPrice`
+will return `🍔{🧅×0, 🧀×0}`. It would be better if it just returned `🍔` to
+indicate that it's a burger without embellishments. One way to handle this is to
+use a ternary expression:
+
+<<< Item.re#ternary{5-7}
+
+Another approach is to pattern match on the tuple of `(lettuce, onions,
+cheese)`:
+
+<<< Item.re#match-tuple{5-7}
+
+However, pattern matching on tuples of more than 2 elements tends to be hard to
+read; it can even be error-prone when some of the elements of the tuple are of
+the same type. For example, what if you accidentally changed the positions of
+the `onions` and `cheese` variables in the second branch:
+
+```reason{3}
+switch (lettuce, onions, cheese) {
+| (false, 0, 0) => {js|🍔|js}
+| (lettuce, cheese, onions) =>
+```
+
+The compiler wouldn't complain but the ensuing logic would likely be wrong.
+
+The best approach here is to use the record itself as the input to the switch
```

Comment:
I can't find our previous thread about this (maybe it was somewhere in this [old PR](https://github.com/melange-re/melange-re.github.io/pull/93)?), but I decided to go with switch expression without the backticks since it's more of a concept than just syntax, same as how I write if-else, ternary, etc without the backticks.

In mainstream languages that have pattern matching (C#, Swift, Java) they've mostly settled on calling this "switch". Python is the only mainstream language I know of which uses "match". But since "switch" is kind of an overloaded term, maybe I should use "pattern match expression" instead?

Reply:
On the other hand, "pattern match expression" seems verbose. Maybe I can define "match expression" as a term in the first place I use it, clarifying that in the rest of the book, it refers to expressions that have pattern matching branches?

Reply:
Good catch, I fixed that old occurrence in https://github.com/melange-re/melange-for-react-devs/pull/21/commits/ad5f95923b2b8cdbc117704b3b8b4434bdb92662